packages/react-composites/src/composites/CallComposite/utils/MediaGalleryUtils.ts

```diff
@@ -0,0 +1,107 @@
+  const locale = useLocale().strings.call;
+  const [announcerString, setAnnouncerString] = useState<string>('');
+  const remoteParticipantsObject = useSelector(getRemoteParticipants);
+  const remoteParticipants = Object.values(remoteParticipantsObject ?? {}).filter((p) => {
+    /**
+     * We need to make sure remote participants are in the connected state. If they are not
+     * they might not have their displayName set in the call state just yet.
+     */
+    if (p.state === 'Connected') {
+      return p;
+    }
+    return;
+  });
+  const [currentParticipants, setCurrentParticipants] = useState<RemoteParticipantState[]>(remoteParticipants);
+
+  useEffect(() => {
+    const setParticipantEventString = (string: string): void => {
+      setAnnouncerString('');
+      setAnnouncerString(string);
+    };
+    if (remoteParticipants.length < currentParticipants.length) {
+      //someone left
+      const whoLeft = currentParticipants.filter((p) => {
+        if (!remoteParticipants.includes(p)) {
+          return p;
+        }
+        return;
+      });
+      setParticipantEventString(
+        createAnnouncmentString(
+          locale.participantLeftNoticeString,
+          locale.unnamedParticipantChangedString,
+          locale.participantsJoinedOverflowString,
+          whoLeft
+        )
+      );
+      setCurrentParticipants(remoteParticipants);
+    } else if (remoteParticipants.length > currentParticipants.length) {
+      // someone joined
+      const whoJoined = remoteParticipants.filter((p) => {
+        if (!currentParticipants.includes(p)) {
+          return p;
+        }
+        return;
+      });
+      setParticipantEventString(
+        createAnnouncmentString(
+          locale.participantJoinedNoticeString,
+          locale.unnamedParticipantChangedString,
+          locale.participantsJoinedOverflowString,
+          whoJoined
+        )
+      );
+      setCurrentParticipants(remoteParticipants);
+    }
+  }, [remoteParticipants, currentParticipants]);
+
+  return announcerString;
+};
+
+/**
+ * Generates the announcement string for when a participant joins or leaves a call.
+ */
+const createAnnouncmentString = (
+  localeString: string,
+  defaultName: string,
+  overflowString: string,
+  participants?: RemoteParticipantState[]
+): string => {
+  if (participants) {
+    console.log(participants);
+    if (participants.length <= 3) {
+      const names = participants.map((p) => p.displayName ?? defaultName).join(', ');
+      return _formatString(localeString, { displayNames: names });
+    } else {
+      const numberOfExtraParticipants = participants.length - 3;
+      const names = participants
+        .slice(0, 2)
+        .map((p) => p.displayName ?? defaultName)
+        .join(', ');
+      const namesPlusExtra =
+        names + _formatString(overflowString, { numOfParticipants: numberOfExtraParticipants.toString() });
+      return _formatString(localeString, { displayNames: namesPlusExtra });
```

Comment:
This string manipulation is hard to understand for someone translating the locale strings.

I'd recommend some redundancy: 
* One locale string that is used for <=3 participants
* A separate locale string that is used for >3 participants, which has slots for the 3 participants as well as a slot for the number of remaining particpants.

You could take it further and claim that we shouldn't assume how participants are concatenated together (i.e, don't assume that `Donald, Prathmesh` is the only way to concatenate names together, and just have different locale string for each case:

```TypeScript
{
  'oneParticipantJoined': '{} has joined the call',
  'twoParticipantsJoined': '{} and {} have joined the call',
  'threeParticipantsJoined': '{}, {} and {} have joined the call',
  'manyParticipantsJoined': '{}, {}, {} and {} others have joined the call',
}
```